src/parser/statement.js

```diff
@@ -150,9 +154,17 @@ module.exports = {
    * ```
    */
   read_inner_statement: function () {
+    let attrs = [];
+    if (this.token === this.tok.T_ATTRIBUTE) {
+      attrs = this.read_attr_list();
+    }
     switch (this.token) {
-      case this.tok.T_FUNCTION:
-        return this.read_function(false, false);
+      case this.tok.T_FUNCTION: {
+        const result = this.read_function(false, false);
+        result.attrGroups = attrs;
+        attrs = [];
```

Comment:
Is the reset of `attrs` really necessary?

Reply:
I believe I have removed all cases where it is not strictly necessary.